pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -161,82 +164,87 @@ def _give_curves_from_samples(
+            and ch_samples.duration == eom_intervals[-1].tf
+        ):
             eom_intervals[-1].tf = total_duration
         # sampling the channel schedule
-        samples = sch.get_samples()
-        extended_samples = samples.extend_duration(total_duration)
-        for slot in sch:
-            if slot.ti == -1:
-                target["initial"] = slot.targets
-                continue
-            else:
-                # If slot is not the first element in schedule
-                if sch.in_eom_mode(slot):
-                    # EOM mode starts
-                    if not in_eom_mode:
-                        in_eom_mode = True
-                        eom_block_n += 1
-                elif in_eom_mode:
-                    # Buffer when EOM mode is disabled and next slot has 0 amp
-                    in_eom_mode = False
-                    if extended_samples.amp[slot.ti] == 0:
-                        eom_end_buffers[eom_block_n] = EOMSegment(
-                            slot.ti, slot.tf
-                        )
-                if (
-                    eom_block_n + 1 < nb_eom_intervals
-                    and slot.tf == eom_intervals[eom_block_n + 1].ti
-                    and extended_samples.det[slot.tf - 1]
-                    == sch.eom_blocks[eom_block_n + 1].detuning_off
-                ):
-                    # Buffer if next is eom and final det matches det_off
-                    eom_start_buffers[eom_block_n + 1] = EOMSegment(
-                        slot.ti, slot.tf
-                    )
+        extended_samples = ch_samples.extend_duration(total_duration)
+
+        for slot in ch_samples.slots:
+            # If slot is not the first element in schedule
+            if ch_samples.in_eom_mode(slot):
+                # EOM mode starts
+                if not in_eom_mode:
+                    in_eom_mode = True
+                    eom_block_n += 1
+            elif in_eom_mode:
+                # Buffer when EOM mode is disabled and next slot has 0 amp
+                in_eom_mode = False
+
+                tf = eom_intervals[eom_block_n].tf
+                if tf is not None and slot.ti != eom_intervals[eom_block_n].tf:
+                    if np.isclose(
+                        extended_samples.amp[
+                            tf
+                            + 2
+                            * cast(
+                                BaseEOM, sampled_seq._ch_objs[ch].eom_config
+                            ).rise_time
+                        ],
+                        0,
+                    ):
+                        eom_end_buffers[eom_block_n] = EOMSegment(tf, slot.ti)
+
+            if (
+                eom_block_n + 1 < nb_eom_intervals
+                and slot.tf == eom_intervals[eom_block_n + 1].ti
+                and np.isclose(
+                    extended_samples.det[slot.tf - 1],
+                    ch_samples.eom_blocks[eom_block_n + 1].detuning_off,
+                    0.006,
```

Comment:
This is the factor introduced to preserve the EOM drawing of start_buffer; I think is one of the last thing to take care of

Reply:
This is less likely to run into the same issue as the EOM end buffer because in general it will be a Pulse, but even that is not guaranteed.